### PR TITLE
119 cstringview non member functions

### LINKED ISSUE #119: CStringView non-member functions
Implement [operator==](https://en.cppreference.com/w/cpp/string/basic_string/operator_cmp.html), [operator<=>](https://en.cppreference.com/w/cpp/string/basic_string/operator_cmp.html) and [std::swap](https://en.cppreference.com/w/cpp/string/basic_string/swap2.html)

### PR DESCRIPTION
**Pull request by dkrutskikh**
### **User description**
close #119

___

### **PR Type**
Enhancement

___

### **Description**
- Add equality (`operator==`) and three-way comparison (`operator<=>`) operators for `CStringView`

- Support comparisons with other `CStringView`, `StringLike` types, and C strings

- Include comprehensive test coverage for all comparison scenarios

- Remove redundant `CStringView` comparison tests from `FixedString` test suite

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["CStringView Class"] --> B["operator== overloads"]
  A --> C["operator<=> overloads"]
  B --> D["CStringView vs CStringView"]
  B --> E["CStringView vs StringLike"]
  B --> F["CStringView vs C string"]
  C --> G["Three-way comparisons"]
  G --> H["std::strong_ordering"]
  I["Test Suite"] --> J["Comprehensive test coverage"]
  J --> K["Edge cases & Unicode"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>c_string_view.hpp</strong><dd><code>Add comparison operator declarations</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

include/core/c_string_view.hpp

<ul><li>Add <code>operator==</code> declarations for all comparison combinations<br> <li> Add <code>operator<=></code> declarations for three-way comparisons<br> <li> Include comprehensive documentation with preconditions and notes<br> <li> Support template overloads for <code>StringLike</code> concept</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/123/files#diff-9d334ca7b15ac828f1536b20c0fcfd89f80a8970841385b4b869e2dd4ae26849">+210/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>c_string_view.inl</strong><dd><code>Implement comparison operator logic</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

include/core/c_string_view.inl

<ul><li>Implement <code>operator==</code> with compile-time and runtime string comparison<br> <li> Implement <code>operator<=></code> returning <code>std::strong_ordering</code><br> <li> Handle edge cases like empty strings and self-comparison<br> <li> Use <code>cstrcmp</code> for constexpr contexts and <code>std::strcmp</code> for runtime</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/123/files#diff-19b0d72c816318d3ee02d11b157d8b518d3c1d3e7eef9964ffae8b3efec84534">+113/-0</a>&nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>c_string_view_test.cpp</strong><dd><code>Add comprehensive comparison operator tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

test/core/c_string_view_test.cpp

<ul><li>Add comprehensive test cases for <code>operator==</code> with various string types<br> <li> Add extensive test cases for <code>operator<=></code> with lexicographic ordering<br> <li> Include tests for edge cases, Unicode, special characters, and <br>performance<br> <li> Add <code>std::swap</code> functionality tests for <code>CStringView</code> objects</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/123/files#diff-2e6a5805b82d6df486ae0d658f5172c7ba8c1a58847a5bb3d7b57e6167c37188">+669/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>fix_string_test.cpp</strong><dd><code>Remove redundant CStringView tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

test/core/fix_string_test.cpp

<ul><li>Remove redundant <code>CStringView</code> comparison tests from <code>FixedString</code> test <br>suite<br> <li> Clean up test sections that were testing <code>CStringView</code> functionality</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/123/files#diff-cc1ff39c1921067e1cd9ffcf0bc8c8f3d7349f0178f1eb6ddeac099bc0a2d2ae">+0/-17</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

File: include/core/c_string_view.inl [hunk 1]
```diff
@@ -469,6 +469,119 @@ constexpr std::size_t CStringView::_find_last_not_of_raw(std::size_t position, c
...
+constexpr bool operator==(const CStringView & lhs, const stringType & rhs) noexcept {
+  if (lhs.c_str() == rhs.c_str() || (lhs.empty() && rhs.empty()))
+    return true;
+
+  if consteval {
+    return cstrcmp(lhs.c_str(), rhs.c_str());
+  } else {
+    return std::strcmp(lhs.c_str(), rhs.c_str()) == 0;
+  }
+}
```
[the remark this answers is not in the document]
Comment (dkrutskikh) [reply include/core/c_string_view.inl:493]: fixed

File: include/core/c_string_view.inl [hunk 2]
```diff
@@ -469,6 +469,119 @@ constexpr std::size_t CStringView::_find_last_not_of_raw(std::size_t position, c
...
+  } else {
+    const int result = std::strcmp(lhs.c_str(), rhs.c_str());
+    if (result < 0)
+      return std::strong_ordering::less;
+    else if (result > 0)
+      return std::strong_ordering::greater;
+    else
+      return std::strong_ordering::equal;
+  }
+}
```
[the remark this answers is not in the document]
Comment (dkrutskikh) [reply include/core/c_string_view.inl:536]: fixed
